iceoryx_posh/source/runtime/posh_runtime.cpp

```diff
@@ -551,6 +588,38 @@ popo::ConditionVariableData* PoshRuntime::getMiddlewareConditionVariable() noexc
     return maybeConditionVariable.value();
 }
 
+popo::EventVariableData* PoshRuntime::getMiddlewareEventVariable() noexcept
+{
+    IpcMessage sendBuffer;
+    sendBuffer << IpcMessageTypeToString(IpcMessageType::CREATE_EVENT_VARIABLE) << m_appName;
+
+    auto maybeEventVariable = requestEventVariableFromRoudi(sendBuffer);
+    if (maybeEventVariable.has_error())
+    {
+        switch (maybeEventVariable.get_error())
+        {
+        case IpcMessageErrorType::EVENT_VARIABLE_LIST_FULL:
+            LogWarn() << "Could not create event variable as we are out of memory for event variables.";
+            errorHandler(Error::kPOSH__RUNTIME_ROUDI_EVENT_VARIABLE_LIST_FULL, nullptr, iox::ErrorLevel::SEVERE);
+            break;
+        case IpcMessageErrorType::REQUEST_EVENT_VARIABLE_WRONG_IPC_MESSAGE_RESPONSE:
+            LogWarn() << "Could not create event variables; received wrong message queue response.";
+            errorHandler(Error::kPOSH__RUNTIME_ROUDI_REQUEST_EVENT_VARIABLE_WRONG_MESSAGE_QUEUE_RESPONSE,
+                         nullptr,
+                         iox::ErrorLevel::SEVERE);
+            break;
+        default:
+            LogWarn() << "Undefined behavior occurred while creating event variable";
+            errorHandler(Error::kPOSH__RUNTIME_ROUDI_EVENT_VARIABLE_CREATION_UNDEFINED_BEHAVIOR,
+                         nullptr,
+                         iox::ErrorLevel::SEVERE);
+            break;
```

Comment:
there are always two log outputs, first a Log::Error from requestEventVariableFromRoudi and then the Log::Warn from here. Please decide if it's error or warning and do the printing only at one place. Maybe by moving everything into requestEventVariableFromRoudi.